packages/sns/lib/sns/AbstractSnsPublisher.ts

```diff
@@ -118,5 +121,29 @@ export abstract class AbstractSnsPublisher<MessagePayloadType extends object>
   override processMessage(): Promise<Either<'retryLater', 'success'>> {
     throw new Error('Not implemented for publisher')
   }
+
+  protected async sendMessage(
+    payload: MessagePayloadType | OffloadedPayloadPointerPayload,
+    options: SNSMessageOptions,
+  ): Promise<void> {
+    const attributes: Record<string, MessageAttributeValue> = {}
+
+    if (isOffloadedPayloadPointerPayload(payload)) {
```

Comment:
this looks suspiciously identical to SQS implementation, maybe we should move it to some util?

Reply:
Fair point
Changed: https://github.com/kibertoad/message-queue-toolkit/pull/167/commits/1e45a4020d0d3e28cfef275ca66d40ac18fca217